### PR TITLE
Include glide deps in vendor

### PR DESCRIPTION
**Pull request by silasdavis**
* Include glide deps in `vendor/`
* Update dependencies to latest masters
* Add testify dep

We may still be missing some deps (haven't looked at test failures exhaustively). Next step is to make tests pass in this branch. We may which to peg some deps to specific version (by tag or git hash) in the glide.yaml to leave them untouched by `glide up`. Ideally we want some dependencies to be upgraded by `glide up` so let's only lock down moving targets.

### CODE REVIEW

**Comment by benjaminbollen** [comment]
I will test this locally to see if we can merge it

**Comment by silasdavis** [comment]
AH.... hold on we have implicit submodules, see https://github.com/Masterminds/glide/issues/112

**Comment by silasdavis** [comment]
Will have to look into this some more, don't merge

File: glide.lock [hunk 1]
```diff
@@ -1,8 +1,95 @@
...
+  subpackages:
+  - erisdb
+  - erisdb/erisdbss
+  - server
+  - erisdb/pipe
+  - rpc
+  - util
+  - files
+  - test/testdata/testdata
+  - Godeps/_workspace/src/github.com/eris-ltd/mint-client/mintx/core
```
Comment (benjaminbollen) [inline glide.lock]: Should this be cleared out?

**Comment by silasdavis** [comment]
@benjaminbollen fixed the tests on this, I'll make another PR to address the data races I've found

File: docs/glide-readme.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+but those tests are likely to break because the the test runner may be unable to find
+their own nested vendored dependencies.
+
+Instead use:
+
+```bash
+glide novendor | xargs go test
+```
+
+Where `glide novendro` returns a newline-delimited list of packages in this project
```
Comment (benjaminbollen) [inline docs/glide-readme.md]: Small typo

**Comment by benjaminbollen** [comment]
Looks very good.

**Pull request merged**